### PR TITLE
add ability to broadcast to other apps

### PR DESCRIPTION
**Pull request by rnpenguin**

### CODE REVIEW

File: app/src/main/java/com/nightscout/android/preferences/AndroidPreferences.java [hunk 1]
```diff
@@ -355,6 +355,10 @@ public boolean isCampingModeEnabled() {
         return preferences.getBoolean(context.getString(R.string.camping_enable), false);
     }
 
+    public boolean isBroadcastEnabled() {
+        return preferences.getBoolean(context.getString(R.string.camping_enable), false);
```
Comment (trhodeos) [inline app/src/main/java/com/nightscout/android/preferences/AndroidPreferences.java]: camping_enable -> broadcast_enabled

File: app/src/main/res/values/pref_keys.xml [hunk 2]
```diff
@@ -32,6 +32,7 @@
     <string name="share_bluetooth_address" translatable="false">share_bluetooth_address</string>
     <string name="bt_scan_share2" translatable="false">bt_scan_share2</string>
     <string name="camping_enable" translatable="false">camping_enable</string>
+    <string name="broadcast_enable" translatable="false">camping_enable</string>
```
Comment (trhodeos) [inline app/src/main/res/values/pref_keys.xml]: camping_enable -> broadcast_enable

File: app/src/main/java/com/nightscout/android/ProcessorService.java [hunk 3]
```diff
@@ -315,6 +315,18 @@ public void incomingData(G4Download download) {
...
         bus.post(response);
+
+        if (preferences.isBroadcastEnabled()) {
+            // TODO: make this work for gap sync or a way to request for data (having a db for this would be great)
+            Intent broadcastIntent = new Intent();
+            broadcastIntent.setAction("com.nightscout.action.PROCESS_RESPONSE");
+            broadcastIntent.addCategory(Intent.CATEGORY_DEFAULT);
+            SensorGlucoseValueEntry sgv = download.sgv.get(download.sgv.size() - 1);
+            EGVRecord currentEGVrecord = new EGVRecord(sgv, rcvrTime, refTime);
+            broadcastIntent.putExtra("nsSgv", currentEGVrecord.getBgMgdl());
```
Comment (jasoncalabrese) [inline app/src/main/java/com/nightscout/android/ProcessorService.java:326]: We should include the units in the field name, `nsSgvMgdl`

**Comment by rnpenguin** [comment]
sounds good, also think NS version, but can add as this becomes more of a model.

**Comment by rnpenguin** [comment]
@ktind

**Comment by ktind** [comment]
:+1:

**Pull request merged**